lib/common/xml.c

```diff
@@ -3787,7 +3787,8 @@ dump_xml_element(xmlNode * data, int options, char **buffer, int *offset, int *m
     if (data->children) {
         xmlNode *xChild = NULL;
 
-        for (xChild = __xml_first_child(data); xChild != NULL; xChild = __xml_next(xChild)) {
+        //for (xChild = __xml_first_child(data); xChild != NULL; xChild = __xml_next(xChild)) {
+        for (xChild = data->children; xChild != NULL; xChild = xChild->next) {
```

Comment:
Rather than comment out the old code, add a comment explaining why __xml_first_child()/__xml_next() aren't appropriate here.